pkg/api/metadata.go

```diff
@@ -36,6 +36,16 @@ func (m *Metadata) IsComplete() error {
 	return nil
 }
 
+// AsString returns a string representation of the metadata suitable for using
+// in human-oriented output
+func (m *Metadata) AsString() string {
```

Comment:
We could implement [`Stringer`](https://pkg.go.dev/fmt#Stringer) instead and drop the method calls.

Reply:
Good as a follow-up